### DISCUSSION TITLE
How easy is it to use custom view rendering?

### DISCUSSION THREAD

**Discussion by jaredcwhite** [jeremyevans/rodauth#301] [General]
I'm starting to evaluate Rodauth as a possible login solution for [Bridgetown](https://www.bridgetownrb.com) sites, and as I understand it Rodauth comes prebuilt with page templates for the login form, etc. As Bridgetown uses its own view template rendering pipeline (and its Roda integration uses that, rather than any native Roda/Tilt rendering), I'm wondering how easy it would be to set up Rodauth to use that. Is there a particular touch point in the gem, or a set of conventions, to make this straightforward? I think the rodauth-rails gem does something conceptually similar where Action View is used for the view layer in Rails.

The other potential hurdle would be getting Sequel set up for Rodauth's DB access, but thanks to @janko's excellent sequel-activerecord_connection gem, it looks like that shouldn't be an issue (currently Bridgetown offers an Active Record plugin, but now we'll be able to support both at once).

**Comment by jeremyevans** [comment]
You would want to override the `view`/`render` Rodauth methods to use Bridgetown's rendering instead of Roda's.  I assume that is what rodauth-rails does, though I haven't looked.

**Reply by janko** [reply]
That's correct, those are the entrypoints overridden by rodauth-rails – https://github.com/janko/rodauth-rails/blob/81faebd14962dd42ca16dad3c05e0fb37f93b4c5/lib/rodauth/rails/feature/render.rb#L11-L25

**Reply by jaredcwhite** [reply]
Thanks @jeremyevans & @janko, figured as such but wanted to double-check my assumptions. 👍
